File: AssignCookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int j=0, res=0;
        for(int i=0; i<g.length && j<s.length; i++){
            while(j<s.length && g[i]>s[j]) j++;
            if(j<s.length && g[i]<=s[j]) {
                res++;
                j++;
            }
        }
        return res;
    }
}

File: AssignCookies.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function(g, s) {
    let j=0, res=0;
    g.sort((a, b) => a-b);
    s.sort((a, b) => a-b);
    for(let i=0; i<g.length; i++){
        while(j<s.length && g[i]>s[j]) j++;
        if(g[i]<=s[j]){
            res++;
            j++;
        }
    }
    return res;
};